homeassistant/components/myq/__init__.py

```diff
@@ -63,3 +73,42 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+class MyQEntity(CoordinatorEntity):
+    """Base class for MyQ Entities."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator, device) -> None:
+        """Initialize class."""
+        super().__init__(coordinator)
+        self._device = device
+        self._attr_unique_id = device.device_id
+
+    @property
+    def name(self):
+        """Return the name if any, name can change if user changes it within MyQ."""
+        return self._device.name
```

Comment:
```suggestion
        self._attr_unique_id = device.device_id
        self._attr_name = device.name
```

Reply:
If I do this then the entity_id ends up being <domain>.unnamed_device

Reply:
Also, device_id will never change as that is the serial number

Reply:
Is the `name` undefined on first load ?

Reply:
No, name would be there but it can change as it is based on the name given to the device by the user. User can change the name through the MyQ app as it is just a label.
As it is now, when the user changes it then the name would end up getting updated here as well and they will thus see it with the new name.
Otherwise it won't change until they restart HASS.